src/de/grobox/liberario/fragments/ProductDialogFragment.java

```diff
@@ -183,6 +186,13 @@ public void bindView(final ViewHolder ui) {
 				public void onClick(View v) {
 					int position = adapter.getAdapterPosition(ProductItem.this);
 					adapter.toggleSelection(position);
+					EnumSet<Product> products = getProductsFromItems(adapter.getSelectedItems());
```

Comment:
I think you don't need to call `getProductsFromItems()` here, just `adapter.getSelectedItems()` should be sufficient.